src/Dsll.hs

```diff
@@ -58,6 +66,16 @@ instance Show Vd where
               cString = show typesVd
               dString = show toVd
 
+-- | Subtype declarations
+data Std = Std {subType :: T,
```

Comment:
`Std` is a confusing name. I would rename it `SubtypeDecl`
(Same for `Od` -> `PredDecl`, etc.)